Hide future payments message from payment element when manual renewal is required

Mayisha opened the pull request:
Fixes STRIPE-460
Fixes #4332 

## Changes proposed in this Pull Request:
- Checking if automatic renewal is disabled for the site. 
- When automatic renewal is disabled and manual renewal is required,
    - Prevent showing the terms stating future usage conditions.
    - Prevent saving the payment method.

## Testing instructions
- Install the WooCommerce Subscriptions plugin and create a subscription product.
- Go to `WooCommerce > Settings > Subscriptions` and disable auto renewal.

<img width="1309" alt="Screenshot 2025-05-21 at 12 07 13 PM" src="https://github.com/user-attachments/assets/a44fca12-6eb8-427e-ada9-e2d5b48b7f1e" />

- As a shopper, add the subscription product to your cart and go to the checkout page. (Test on both shortcode and block checkout)
- Confirm that the Stripe terms about saving the payment method for future usage are not displayed.

Shortcode Checkout | Block Checkout
--- | ---
<img width="450" alt="Screenshot 2025-05-23 at 3 36 33 PM" src="https://github.com/user-attachments/assets/771b4e2d-3ea0-4647-b856-ea27684f52d6" /> | <img width="766" alt="Screenshot 2025-05-23 at 3 36 11 PM" src="https://github.com/user-attachments/assets/52eeb164-c62b-4e55-ab68-81e1e7dc6702" />

- Go to `My Account > Payment methods` page and confirm that the card has not been saved.
- Go to `My Account > Subscriptions` and view the subscription. Confirm that the subscription is set to manual renewal.
- Regression test
    - Enable automatic payment.
    - As a shopper, purchase a subscription.
    - Confirm that on the checkout page, the Stripe terms about saving the payment method for future usage are displayed.

    <img width="460" alt="Screenshot 2025-05-23 at 3 24 45 PM" src="https://github.com/user-attachments/assets/9f94a3ce-93e0-457e-a910-e039f50ec235" />

    - Confirm that the card is saved in `My Account > Payment methods` page.

Mayisha committed:
add helper method to check if manual renewal is required for subscriptions

Mayisha committed:
hide future usage terms from stripe element when manual renewal is required in the store

Mayisha committed:
add changelog

Mayisha committed:
remove debug log

Mayisha committed:
only save card when automatic renewal is allowed

Mayisha committed:
Merge branch 'develop' into task/4332-hide-terms-for-manual-renew

malithsen committed:
Merge branch 'develop' into task/4332-hide-terms-for-manual-renew

malithsen commented on client/blocks/upe/upe-deferred-intent-creation/payment-elements.js:
```diff
@@ -155,7 +155,9 @@ const PaymentElements = ( {
-				getBlocksConfiguration()?.cartContainsSubscription ||
+				( getBlocksConfiguration()?.cartContainsSubscription &&
+					! getBlocksConfiguration()
+						?.subscriptionRequiresManualRenewal ) ||
```
Maybe extract this into a new function to make it more readable

```
const shouldSetupOffSessionPayment = () => {
    const config = getBlocksConfiguration();
    const hasAutoRenewingSubscription = config?.cartContainsSubscription && 
        !config?.subscriptionRequiresManualRenewal;
    
    return hasAutoRenewingSubscription || props.showSaveOption;
};

if ( shouldSetupOffSessionPayment() ) {
    // ...
}
```

Mayisha replied:
Done in [ef230da](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4340/commits/ef230dab476d1bbaa96f5d706cf92055c4b2643c)

malithsen reviewed (changes requested):
Thanks for working on this fix @Mayisha  👍 

Noticed an issue when testing.
On shortcode checkout, regardless of manual/auto renewal the notice is always displayed.

![CleanShot 2025-05-27 at 14 50 32](https://github.com/user-attachments/assets/988b86f5-2708-4896-a88a-60d7beed6c97)

Mayisha committed:
Merge branch 'develop' into task/4332-hide-terms-for-manual-renew

Mayisha committed:
extract logic to a util function

Mayisha commented:
Thanks @malithsen. I have addressed the feedback about the code extraction to a separate function. 
However, I am unable to reproduce the shortcode issue. Could you share some more details about your subscription renewal setting, Stripe settings etc? 👀

malithsen reviewed (approved):
> However, I am unable to reproduce the shortcode issue. Could you share some more details about your subscription renewal setting, Stripe settings etc? 👀

Hm, I am not able to replicate this anymore either 🤷 

I tested legacy, new and optimized checkout with logged in logged out users. Both Block and shortcode looks good to me now.

Mayisha committed:
Merge branch 'develop' into task/4332-hide-terms-for-manual-renew
